sympy/functions/elementary/piecewise.py

```diff
@@ -305,13 +306,115 @@ def doit(self, **hints):
             newargs.append((e, c))
         return self.func(*newargs)
 
+    def is_meromorphic(self, x, a):
+        """
+        This method tests whether a Piecewise expression is meromorphic
+        as a function of the given symbol ``x`` at the point ``a``.
+
+        This method is intended as a quick test that will return
+        None if no decision can be made without simplification or
+        more detailed analysis.
+
+        Examples
+        ========
+        >>> from sympy import csc, log, Piecewise
+        >>> from sympy.abc import x
+        >>> f = Piecewise((csc(1/x), x < 1), (2*x**2 + 1, x < 5), (log(x), True))
+        >>> f.is_meromorphic(x, 0)
+        False
+        >>> f.is_meromorphic(x, 2)
+        True
+        >>> f.is_meromorphic(x, 10)
+        True
+
+        """
+        if not x.is_symbol:
+            raise TypeError("{} should be of symbol type".format(x))
+        a = sympify(a)
+        upper_bound_list, lower_bound_list = [], []
+        for l, u, e, i in self._intervals(x)[1]:
+            upper_bound_list.append(u)
+            lower_bound_list.append(l)
+
+        if a not in lower_bound_list and a not in upper_bound_list:
+            for bound in self.as_expr_set_pairs():
+                if a in bound[1]:
+                    return bound[0]._eval_is_meromorphic(x, a)
+        else:
+            return None
+
     def _eval_simplify(self, **kwargs):
         return piecewise_simplify(self, **kwargs)
 
-    def _eval_as_leading_term(self, x, logx=None, cdir=0):
-        for e, c in self.args:
-            if c == True or c.subs(x, 0) == True:
-                return e.as_leading_term(x)
+    def _eval_as_leading_term(self, x, logx=None, cdir=1):
+        from sympy.sets.sets import Union
+        if cdir == S.Zero:
+            l = self._eval_as_leading_term(x, logx=None, cdir=-1)
+            r = self._eval_as_leading_term(x, logx=None, cdir=1)
+            if l != r:
+                raise ValueError
+            return r
+
+        upper_bound_list, lower_bound_list = [], []
+        for a, b, e, i in self._intervals(x)[1]:
+            upper_bound_list.append(b)
+            lower_bound_list.append(a)
```

Comment:
Why do you collect all bounds? It should suffice to take the first interval that contains the origin, possibly as an endpoint depending on `cdir`.

Reply:
Yeah , taking all bounds isn't super necessary here , will try to improve on that !